engine/table/src/main/java/io/deephaven/engine/table/impl/by/ChunkedOperatorAggregationHelper.java

```diff
@@ -86,34 +86,14 @@ public static QueryTable aggregation(
             final boolean preserveEmpty,
             @Nullable final Table initialKeys,
             @NotNull final Collection<? extends ColumnName> groupByColumns) {
-        final String[] keyNames = groupByColumns.stream().map(ColumnName::name).toArray(String[]::new);
-        if (!input.hasColumns(keyNames)) {
-            final Set<String> colNames = input.getColumnSourceMap().keySet();
-            final String[] missingColumns = Arrays.stream(keyNames)
-                    .filter(Predicate.not(colNames::contains))
-                    .toArray(String[]::new);;
-
-            throw new IllegalArgumentException("aggregation: not all group-by columns " + Arrays.toString(keyNames)
-                    + " are present in input table with columns "
-                    + Arrays.toString(input.getDefinition().getColumnNamesArray()) + ". Missing columns: "
-                    + Arrays.toString(missingColumns));
-        }
+        final List<String> keyNames = groupByColumns.stream().map(ColumnName::name).collect(Collectors.toList());
+        input.checkAvailableColumns(keyNames);
         if (initialKeys != null) {
-            if (keyNames.length == 0) {
+            if (keyNames.isEmpty()) {
                 throw new IllegalArgumentException(
                         "aggregation: initial groups must not be specified if no group-by columns are specified");
             }
-            if (!initialKeys.hasColumns(keyNames)) {
-                final Set<String> colNames = input.getColumnSourceMap().keySet();
-                final String[] missingColumns = Arrays.stream(keyNames)
-                        .filter(Predicate.not(colNames::contains))
-                        .toArray(String[]::new);;
-
-                throw new IllegalArgumentException("aggregation: not all group-by columns " + Arrays.toString(keyNames)
-                        + " are present in initial groups table with columns "
-                        + Arrays.toString(initialKeys.getDefinition().getColumnNamesArray()) + ". Missing columns: "
-                        + Arrays.toString(missingColumns));
-            }
+            initialKeys.getDefinition().checkColumns(keyNames);
```

Comment:
Same comment regarding specialization of the error message. Using `checkAvailableColumns` is less user-friendly, especially when there are two inputs to be concerned with.